tests/core/data/layer/v1/fs/test_repository.py

```diff
@@ -0,0 +1,180 @@
+import os
+import pickle
+import shutil
+import unittest
+from glob import glob
+from io import BytesIO
+from pathlib import Path
+from typing import Dict, Any, List
+
+import pandas as pd
+
+from py4ai.core.data.layer.v1.common.executors import AsyncExecutor
+from py4ai.core.data.layer.v1.common.repository import QueryOptions, SortingDirection
+from py4ai.core.data.layer.v1.fs.criteria import (
+    FileSystemCriteriaFactory,
+    FileSystemSearchCriteria,
+)
+from py4ai.core.data.layer.v1.fs.repository import FileSystemRepository
+from py4ai.core.data.layer.v1.fs.serializer import (
+    IndexedIO,
+    FileSerializerMode,
+    FileSerializer,
+)
+from tests import DATA_FOLDER, TMP_FOLDER
+from tests.core.data.layer.v1.base import Entity, EntityRepository, CriteriaFactory
+
+
+class PickleEntitySerializer(FileSerializer[int, Entity]):
+    mode = FileSerializerMode.BINARY
+
+    def get_key(self, entity: Entity) -> int:
+        return entity.cai
+
+    def to_document_key(self, key: int) -> str:
+        return os.path.join(self.path, str(key) + ".pkl")
+
+    def to_model(self, document: IndexedIO) -> Entity:
+        return pickle.load(document.buffer)
+
+    def to_document(self, entity: Entity) -> IndexedIO:
+        buffer = BytesIO()
+        buffer.write(pickle.dumps(entity, protocol=pickle.HIGHEST_PROTOCOL))
+        buffer.seek(0)
+        return IndexedIO(name=self.get_key(entity), buffer=buffer)
+
+
+class FileSystemEntityCriteriaFactory(
+    CriteriaFactory[List[int]], FileSystemCriteriaFactory[int, Entity]
+):
+    def get_index_fields(self, entity: Entity) -> Dict[str, Any]:
+        return {"cai": entity.cai, "birth_year": entity.birth_year}
+
+    def by_cai(self, cai: int) -> FileSystemSearchCriteria:
+        return self.filter_path_by_condition(lambda data: data["cai"] == cai)
+
+    def from_birth_year(self, birth_year: int) -> FileSystemSearchCriteria:
+        return self.filter_path_by_condition(
+            lambda data: data["birth_year"] >= birth_year
+        )
+
+    def by_birth_year(self, birth_year: int) -> FileSystemSearchCriteria:
+        return self.filter_path_by_condition(
+            lambda data: data["birth_year"] == birth_year
+        )
+
+
+class PickleEntityRepository(
+    FileSystemRepository[int, Entity],
+    EntityRepository[str, Dict[str, Any], List[int]],
+):
+    criteria: FileSystemEntityCriteriaFactory
+
+    def __init__(self, path: Path, serializer: PickleEntitySerializer):
+        super().__init__(path, serializer)
+        self.criteria = FileSystemEntityCriteriaFactory(path)
```

Comment:
Why creating this objects here instead of using the homologous objects defined in the package?
This way aren't you testing something different w.r.t. you implemented in the package?

Reply:
Not sure I fully understand what you mean. Can you clarify?

Reply:
Nevermind. I now got the reason behind these classes.